File: strands-py/backend/src/infrastructure/team_context.py
"""team_context と Skills / Workflows ルートのテーブル駆動解決。"""

from __future__ import annotations

import logging
from pathlib import Path
from typing import Any

from src.infrastructure.workspace import STRANDS_PY_ROOT

logger = logging.getLogger(__name__)

TEAM_CONTEXT_DEVELOPER = "developer"
VALID_TEAM_CONTEXTS: frozenset[str] = frozenset(
    {
        TEAM_CONTEXT_DEVELOPER,
        "marketing",
        "hr",
        "pmo",
        "engineering",
    }
)

team_context_labels: dict[str, str] = {
    TEAM_CONTEXT_DEVELOPER: "Developer",
    "marketing": "Marketing Teams",
    "hr": "HR Teams",
    "pmo": "PMO Teams",
    "engineering": "Engineering Teams",
}


def normalize_team_context(raw: Any) -> str:
    if raw is None:
        return TEAM_CONTEXT_DEVELOPER
    if not isinstance(raw, str):
        logger.warning("team_context が文字列ではありません: %r → developer に正規化", raw)
        return TEAM_CONTEXT_DEVELOPER
    s = raw.strip().lower()
    if not s or s not in VALID_TEAM_CONTEXTS:
        if s:
            logger.warning("未知の team_context %r → developer に正規化", raw)
        return TEAM_CONTEXT_DEVELOPER
    return s


def resolve_skills_root(team_context: str) -> Path:
    base = STRANDS_PY_ROOT / "backend" / "agent" / "skills"
    if team_context == TEAM_CONTEXT_DEVELOPER:
        return base / "dev"
    return base / "pr"


def resolve_workflows_root() -> Path:
    return STRANDS_PY_ROOT / "backend" / "agent" / "workflows" / "dev"


def workflow_enabled(team_context: str) -> bool:
    return team_context == TEAM_CONTEXT_DEVELOPER

File: strands-py/frontend/src/lib/teamContext.ts
/**
 * team_context ID と表示ラベルは backend の
 * src/infrastructure/team_context.py（VALID_TEAM_CONTEXTS / team_context_labels）と同期すること。
 */

export type TeamContextId =
  | 'developer'
  | 'marketing'
  | 'hr'
  | 'pmo'
  | 'engineering'

export const TEAM_CONTEXT_OPTIONS: { id: TeamContextId; label: string }[] = [
  { id: 'developer', label: 'Developer' },
  { id: 'marketing', label: 'Marketing Teams' },
  { id: 'hr', label: 'HR Teams' },
  { id: 'pmo', label: 'PMO Teams' },
  { id: 'engineering', label: 'Engineering Teams' },
]

export function workflowEnabled(teamContext: string): boolean {
  return teamContext === 'developer'
}

const LS_ACK_TEAM_PR = 'strands_ack_team_pr_v1'

export function hasAcknowledgedTeamPrNotice(): boolean {
  try {
    return localStorage.getItem(LS_ACK_TEAM_PR) === '1'
  } catch {
    return false
  }
}

export function setAcknowledgedTeamPrNotice(): void {
  try {
    localStorage.setItem(LS_ACK_TEAM_PR, '1')
  } catch {
    /* ignore */
  }
}
